recipes-openxt/tss/tpm2.0-tools/0001-tpm2-tools-lib-support.patch

```diff
@@ -0,0 +1,380 @@
++    TPM2B_ENCRYPTED_SECRET  encryptedSalt = { {0}, };
++    TPMT_SYM_DEF symmetric;
++    TPM_RC rval;
++    TPM2B_NONCE nonceCaller;
++
++    nonceCaller.t.size = 0;
++    policyDigest.t.size = 0;
++
++    // Start policy session.
++    symmetric.algorithm = TPM_ALG_NULL;
++    rval = tpm_session_start_auth_with_params(sysContext, policySession, TPM_RH_NULL, 0, TPM_RH_NULL, 0, &nonceCaller, &encryptedSalt,
++        trial ? TPM_SE_TRIAL : TPM_SE_POLICY, &symmetric, nameAlg);
++    if( rval != TPM_RC_SUCCESS )
++    {
++        printf("build_policy_external, Unable to Start Auth Session, Error Code: 0x%x\n", rval);
++        return rval;
++    }
++
++    // Send policy command.
++    rval = build_pcr_policy( sysContext, *policySession, pcrList, pcrCount, nameAlg);
++    if( rval != TPM_RC_SUCCESS )
++    {
++        printf("build_pcr_policy, Error Code: 0x%x\n", rval);
++        return rval;
++    }
++
++    // Get policy hash.
++    INIT_SIMPLE_TPM2B_SIZE( policyDigest );
++    rval = Tss2_Sys_PolicyGetDigest( sysContext, (*policySession)->sessionHandle,
++            0, &policyDigest, 0 );
++    if( rval != TPM_RC_SUCCESS )
++    {
++        printf("PolicyGetDigest, Error Code: 0x%x\n", rval);
++        return rval;
++    }
++
++    if( trial )
++    {
++        // Need to flush the session here.
++        rval = Tss2_Sys_FlushContext( sysContext, (*policySession)->sessionHandle );
++        if( rval != TPM_RC_SUCCESS )
++            return rval;
++
++        // And remove the session from sessions table.
++        rval = tpm_session_auth_end( *policySession );
++        if( rval != TPM_RC_SUCCESS )
++            return rval;
++    }
++
++    memcpy(policyDigestOut->t.buffer, policyDigest.t.buffer, policyDigest.t.size);
++    policyDigestOut->t.size = policyDigest.t.size;
++    return rval;
++
++}
++
+diff --git a/lib/context-util.c b/lib/context-util.c
+index 7b9a454..d61e913 100644
+--- a/lib/context-util.c
++++ b/lib/context-util.c
+@@ -142,7 +142,7 @@ sapi_ctx_init (TSS2_TCTI_CONTEXT *tcti_ctx)
+         .tssVersion = TSS_SAPI_FIRST_VERSION,
+     };
+ 
+-    size = Tss2_Sys_GetContextSize (0);
++    size = Tss2_Sys_GetContextSize (12000);
+     sapi_ctx = (TSS2_SYS_CONTEXT*)calloc (1, size);
+     if (sapi_ctx == NULL) {
+         fprintf (stderr,
+diff --git a/lib/pcr.c b/lib/pcr.c
+index d781ab5..f5d719b 100644
+--- a/lib/pcr.c
++++ b/lib/pcr.c
+@@ -7,7 +7,7 @@
+ #include "pcr.h"
+ #include "string-bytes.h"
+ 
+-static int pcr_get_id(const char *arg, UINT32 *pcrId)
++int pcr_get_id(const char *arg, UINT32 *pcrId)
+ {
+     UINT32 n = 0;
+ 
+@@ -61,6 +61,25 @@ static int pcr_parse_selection(const char *str, int len, TPMS_PCR_SELECTION *pcr
+     return 0;
+ }
+ 
++int pcr_parse_arg(char *arg, UINT32 *pcrId, BYTE *forwardHash)
++{
++    char * pstr;
++    UINT16 length;
++    int ret = 0;
++    if (strchr(arg, ':')) {
++        //read forward hash and convert to hex to byte
++        pstr = strtok(arg, ":");
```

Comment:
I did a little digging and while I did not find documentation clearly stating it was acceptable, I did find multiple code bases (one notable example being BerkleyDB) where they strtok'ed optarg. I think this will be okay.